apollo/Objective.tsx

```diff
@@ -157,6 +171,46 @@ export function checkObjectives(
     gameState.push(...newGameState);
   }
 
+  if (deniedOptionalCondition) {
+    const denyingPlayer = pickWinningPlayer(
+      previousMap,
+      activeMap,
+      lastActionResponse,
+      deniedOptionalCondition,
+    );
+    let deniedPlayers = (
+      deniedOptionalCondition.players &&
+      deniedOptionalCondition.players.length > 0
+        ? deniedOptionalCondition.players
+        : map.active
+    )?.filter((playerId) => playerId !== denyingPlayer);
+
+    if (
+      deniedPlayers.some(
+        (player) => !deniedOptionalCondition.failed?.has(player),
+      )
+    ) {
+      deniedPlayers = deniedPlayers.filter(
+        (player) => !deniedOptionalCondition.failed?.has(player),
+      );
```

Comment:
Is this the part that you are unsure about?

Reply:
Yeah I'm not entirely sure whether defining `deniedPlayers` based on `condition.players` or `map.active` is the right move. I think it'd be best to identify exactly who was denied by a destructive action, but I guess that information is hard to get in the current structure?